persistence/genesis.go

```diff
@@ -0,0 +1,371 @@
+package persistence
+
+import (
+	"encoding/hex"
+	"fmt"
+	"github.com/pokt-network/pocket/persistence/schema"
+	"github.com/pokt-network/pocket/shared/types/genesis"
+	"github.com/pokt-network/pocket/utility/types"
+	"log"
+)
+
+func (pm *persistenceModule) PopulateGenesisState(state *genesis.GenesisState) { // TODO (Andrew) genericize with actors interface once merged with #111
+	log.Println("Populating genesis state...")
```

Comment:
In my previous implementation in populating the genesis DB I had a couple things that I don't see here.

1. Creating an account for each value
2. Using the genesis stake/value for all validators to determine the value in the pool (for consistency)

I copy-pasted a small code snippet below for reference but wanted to know whether this business logic is incorrect?

```go
        poolValues := make(map[string]*big.Int, 0)

	addValueToPool := func(poolName string, valueToAdd string) error {
		value, err := types.StringToBigInt(valueToAdd)
		if err != nil {
			return err
		}
		if present := poolValues[poolName]; present == nil {
			poolValues[poolName] = big.NewInt(0)
		}
		poolValues[poolName].Add(poolValues[poolName], value)
		return nil
	}

	for _, v := range state.Validators {
		// DISCUSS_IN_THIS_COMMIT(drewskey): If the validator (and other actors) have a balance, do we expect that
		// to be in the `accounts` type corresponding to the same address?
		if err := ctx.SetAccountAmount(v.Address, "0"); err != nil {
			return err
		}
		if err := ctx.InsertValidator(v.Address, v.PublicKey, v.Output, v.Paused, int(v.Status), v.ServiceUrl, v.StakedTokens, v.PausedHeight, v.UnstakingHeight); err != nil {
			return err
		}
		if err := addValueToPool(genesis.ValidatorStakePoolName, v.StakedTokens); err != nil {
			return err
		}
	}
```

Reply:
The pool piece makes sense, but the account piece doesn't matter. Validators are not expected to have accounts associated with them just because they exist

Reply:
It's also expected in some genesis files for this logic to be pre-populated by the creator... That's why the pool field will likely exist in the genesis object

Reply:
Added with [d2edd36](https://github.com/pokt-network/pocket/pull/140/commits/d2edd36ddba729097db8ddbe6b82a26a3f64c4f4)